sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/HyperLogLogPlusPlusHelper.scala

```diff
@@ -93,6 +94,8 @@ class HyperLogLogPlusPlusHelper(relativeSD: Double) extends Serializable {
     val value = dataType match {
       case FloatType => FLOAT_NORMALIZER.apply(_value)
       case DoubleType => DOUBLE_NORMALIZER.apply(_value)
+      case _: StringType => CollationFactory.getCollationKeyBytes(
+        _value.asInstanceOf[UTF8String], dataType.asInstanceOf[StringType].collationId)
```

Comment:
Let's try to avoid `asInstanceOf`. For this particular case, we can use `case s: StringType`, where we can extract information about collations from `s`.